src/huggingface_hub/commands/jobs/uv.py

```diff
@@ -0,0 +1,223 @@
+        run_parser.add_argument("script_args", nargs="...", help="Arguments for the script", default=[])
+        run_parser.add_argument(
+            "--repo",
+            help="Repository name for the script (creates ephemeral if not specified)",
+        )
+        run_parser.add_argument("--flavor", type=str, default="cpu-basic", help="Hardware flavor (default: cpu-basic)")
+        run_parser.add_argument("-e", "--env", action="append", help="Environment variables")
+        run_parser.add_argument("-s", "--secret", action="append", help="Secret environment variables")
+        run_parser.add_argument("--env-file", type=str, help="Read in a file of environment variables.")
+        run_parser.add_argument(
+            "--secret-env-file",
+            type=str,
+            help="Read in a file of secret environment variables.",
+        )
+        run_parser.add_argument("--timeout", type=str, help="Max duration (e.g., 30s, 5m, 1h)")
+        run_parser.add_argument("-d", "--detach", action="store_true", help="Run in background")
+        run_parser.add_argument("--token", type=str, help="HF token")
+        # UV options
+        run_parser.add_argument("--with", action="append", help="Run with the given packages installed", dest="with_")
+        run_parser.add_argument(
+            "-p", "--python", type=str, help="The Python interpreter to use for the run environment"
+        )
+        run_parser.set_defaults(func=UvCommand)
+
+    def __init__(self, args):
+        """Initialize the command with parsed arguments."""
+        self.args = args
+
+    def run(self):
+        """Execute UV command."""
+        if self.args.uv_command == "run":
+            self._run_script(self.args)
+
+    def _run_script(self, args):
+        """Run a UV script on HF infrastructure."""
+        print("Note: huggingface-cli jobs uv run is experimental and subject to change.")
+        api = HfApi(token=args.token)
+
+        if args.script.startswith("http://") or args.script.startswith("https://"):
+            # Direct URL execution - no upload needed
+            script_url = args.script
+            print(f"Running script from URL: {script_url}")
+        else:
+            # Local file - upload to HF
+            script_path = Path(args.script)
+            if not script_path.exists():
+                print(f"Error: Script not found: {args.script}")
+                return
+
+            # Determine repository
+            repo_id = self._determine_repository(args, api)
+            is_ephemeral = args.repo is None
+
+            # Create repo if needed
+            try:
+                api.repo_info(repo_id, repo_type="dataset")
+                if not is_ephemeral:
+                    print(f"Using existing repository: {repo_id}")
+            except RepositoryNotFoundError:
+                print(f"Creating repository: {repo_id}")
+                create_repo(repo_id, repo_type="dataset", exist_ok=True)
```

Comment:
```suggestion
                create_repo(repo_id, repo_type="dataset", private=True, exist_ok=True)
```
to not risk leaking stuff...